src/providers/network-connection-provider.test.tsx

```diff
@@ -0,0 +1,202 @@
+import { NetworkConnectionProvider } from "./network-connection-provider";
+import {
+    NetworkConnection,
+    NetworkInformationUtils,
+} from "andculturecode-javascript-core";
+import { useNetworkConnection } from "../hooks/use-network-connection";
+import { NetworkConnectionFactory } from "../tests/factories/network-connection-factory";
+
+// -----------------------------------------------------------------------------------------
+// #region Types
+// -----------------------------------------------------------------------------------------
+
+type TypeFromKey<Type, Key extends keyof Type> = Type[Key];
+
+// #endregion Types
+
+// -----------------------------------------------------------------------------------------
+// #region Interfaces
+// -----------------------------------------------------------------------------------------
+
+interface SetupSutOptions {
+    mockConnections?: Array<Partial<NetworkConnection>>;
+}
+
+interface SetupSutResults {
+    TestComponent: () => JSX.Element;
+    networkConnectionResults: {
+        all: Array<NetworkConnection>;
+        current?: NetworkConnection;
+    };
+}
+
+// #endregion Interfaces
+
+// -----------------------------------------------------------------------------------------
+// #region Setup
+// -----------------------------------------------------------------------------------------
+
+const getNetworkConnectionMock = jest.spyOn(
+    NetworkInformationUtils,
+    "getNetworkConnection"
+);
+
+const setupMocks = (mockConnections: Array<Partial<NetworkConnection>>) => {
+    getNetworkConnectionMock.mockReset();
+    for (let index = 0; index < mockConnections.length; index++) {
+        const mockImplementation =
+            index === mockConnections.length - 1
+                ? getNetworkConnectionMock.mockImplementation
+                : getNetworkConnectionMock.mockImplementationOnce;
+
+        mockImplementation(() => {
+            return {
+                isOnline: true,
+                ...mockConnections[index],
+            };
+        });
+    }
+};
+
+const setupSut = (options?: SetupSutOptions): SetupSutResults => {
+    const { mockConnections = [] } = options ?? {};
+
+    setupMocks(mockConnections);
+
+    const networkConnectionResults: TypeFromKey<
+        SetupSutResults,
+        "networkConnectionResults"
+    > = {
+        all: [] as NetworkConnection[],
+    };
+
+    function TestComponent() {
+        const connection = useNetworkConnection();
+        networkConnectionResults.all.push(connection);
+        networkConnectionResults.current = connection;
+
+        return <div></div>;
+    }
+
+    return {
+        TestComponent,
+        networkConnectionResults,
+    };
+};
+
+// #endregion Setup
+
+describe("NetworkConnectionProvider", () => {
+    it("renders initial network connection state", () => {
+        // Arrange
+        const expectedConnection = NetworkConnectionFactory.build();
```

Comment:
Ahhh 😛 I was hoping this could be referenced with `Factory.build<NetworkConnection>(FactoryType.NetworkConnection)` - should be able to now that it's exported from the index. You'll want to add the `FactoryType` value `src/tests/factories/factory-type.ts` too

Reply:
Ah.  I see.